src/System.Private.CoreLib/shared/System/Text/Rune.cs

```diff
@@ -1078,7 +1085,7 @@ public static UnicodeCategory GetUnicodeCategory(Rune value)
         {
             if (value.IsAscii)
             {
-                return (UnicodeCategory)(AsciiCharInfo[value.Value] & UnicodeCategoryMask);
+                return (UnicodeCategory)(AsciiCharInfo[(byte)value.Value] & UnicodeCategoryMask);
```

Comment:
The padding of the array and casting to byte is unnatural code. I do not think we should not be encouraging unnatural code like this.

It would be better to make general bound checks elimination work. E.g. if this was rewritten like this:

```
if (value._value < (uint)AsciiCharInfo.Length)
{
    return (UnicodeCategory)(AsciiCharInfo[value._value] & UnicodeCategoryMask);
}
else
...
```

The JIT should be able to figure out that the redundant bounds check is not necessary.

Reply:
I agree, that was more like an example.